File: core/src/main/java/com/canyon/inject/InjectProvider.java
package com.canyon.inject;

/**
 * 对于没有实例化的Bean，需要注入，需要实现此接口
 * 如：Config配置
 */
public interface InjectProvider {
    boolean isMatch(Class<?> clazz);

    <T> T create(Class<T> objClazz, Class<T> fieldClazz);
}

File: core/src/main/kotlin/com/canyon/inject/InjectProvider.kt
package com.canyon.inject

import kotlin.reflect.KClass

/**
 * 对于没有实例化的Bean，需要注入，需要实现此接口
 * 如：Config配置
 */
interface InjectProvider {
    fun isMatch(kClass: KClass<*>): Boolean
    fun create(objClass: KClass<*>, fieldClass: KClass<*>): Any
}